test_journal.py

```diff
@@ -0,0 +1,279 @@
+    # flush the session to "write" the data to the database
+    db_session.flush()
+    # now, we should have one entry:
+    assert db_session.query(journal.Entry).count() == 1
+    for field in kwargs:
+        if field != 'session':
+            assert getattr(entry, field, '') == kwargs[field]
+    # id and created should be set automatically upon writing to db:
+    for auto in ['id', 'created']:
+        assert getattr(entry, auto, None) is not None
+
+
+def test_entry_no_title_fails(db_session):
+    bad_data = {'text': 'test text'}
+    journal.Entry.write(session=db_session, **bad_data)
+    with pytest.raises(IntegrityError):
+        db_session.flush()
+
+
+def test_entry_no_text_fails(db_session):
+    bad_data = {'title': 'test title'}
+    journal.Entry.write(session=db_session, **bad_data)
+    with pytest.raises(IntegrityError):
+        db_session.flush()
+
+
+def test_read_entries_empty(db_session):
+    entries = journal.Entry.all()
+    assert len(entries) == 0
+
+
+def test_read_entries_one(db_session):
+    title_template = "Title {}"
+    text_template = "Entry Text {}"
+    # write three entries, with order clear in the title and text
+    for x in range(3):
+        journal.Entry.write(
+            title=title_template.format(x),
+            text=text_template.format(x),
+            session=db_session)
+        db_session.flush()
+    entries = journal.Entry.all()
+    assert len(entries) == 3
+    assert entries[0].title > entries[1].title > entries[2].title
+    for entry in entries:
+        assert isinstance(entry, journal.Entry)
+
+
+def test_empty_listing(app):
+    response = app.get('/')
+    assert response.status_code == 200
+    actual = response.body
+    expected = 'No entries here so far'
+    assert expected in actual
+
+
+def test_listing(app, entry):
+    response = app.get('/')
+    assert response.status_code == 200
+    actual = response.body
+    for field in ['title']:
+        expected = getattr(entry, field, 'absent')
+        assert expected in actual
+
+
+def test_about_view(app):
+    response = app.get('/about')
+    assert response.status_code == 200
+    actual = response.body
+    expected = '<h1>Learning Journal</h1>'
+    assert expected in actual
+
+
+def test_create_view_redirect_then_login(app):
+    username, password = ('admin', 'secret')
+    redirect = login_helper(username, password, app)
+    assert redirect.status_code == 302
+    response = redirect.follow()
+    assert response.status_code == 200
+    # Now check logged in view
+    response = app.get('/create')
+    assert response.status_code == 200
+    actual = response.body
+    expected = '<h3>New Entry</h3>'
+    assert expected in actual
+
+
+def test_edit_view_redirect(app):
+    redirect = app.get('/edit/1')
+    assert redirect.status_code == 302
+    response = redirect.follow()
+    assert response.status_code == 200
+    actual = response.body
+    expected = "<h3>Login</h3>"
+    assert expected in actual
```

Comment:
I meant to do something similar with the test_create_view_redirect_then_login, but I think I forgot to come back to it.